README.md

```diff
@@ -1237,3 +1237,32 @@ DELETE /api/v1/users/mary
 ```
 HTTP/1.1 200 OK
 ```
+
+# Upgrades
+
+Initiates an upgrade to the provided version. Once this endpoint is called [Horticulturalist](https://github.com/medic/horticulturalist) will take over and complete the installation.
+
+### Permissions
+
+`_admin` (CouchDB Administrator)
+
+## Example
+
+```
+POST /api/v1/upgrade
+{
+  "build": {
+    "namespace": "medic",
+    "application": "medic",
+    "version": "3.0.0-beta.1"
+  }
+}
+```
+
```

Comment:
OK I think I'm going to leave this as is for now.

I think that you're suggestion Milan is a possible correct implementations of this api.

But there are quite a few possible approaches we could take, and this is not a discussion we've had yet. I don't think there is value to moving to it without having these discussions, as there are definite downsides to it, which I half mentioned above[1]. I don't think it would help to simplify the code we already have either unfortunately, because we'd still need to hardcode our understanding of these various urls and version concepts in, to make sure people are only using it in the locked down way that we currently support. 

We can definitely change this later. If we stick with Horti and want to use it with other medic apps we *will* change it later. Now is just not the time.

[1] One obvious specific is the implication that you can upgrade a deployed app from any build server, because you're passing it on every upgrade. That seems really unsafe to me, unless we supported signing packages or something, because it removes ownership of our releases from us. I would favour configuring the build server in another location. There are probably more to consider, the overall point is that the consideration hasn't been done yet